### PR TITLE
bpo-42018: winreg SetValue(Ex) should mention integer as an acceptable value

### PR DESCRIPTION
**Pull request by kainjow**
https://bugs.python.org/issue42018

https://bugs.python.org/issue42018

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept this contribution by verifying everyone involved has signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

## Recognized GitHub username

We couldn't find a [bugs.python.org](https://bugs.python.org/) (b.p.o) account corresponding to the following GitHub usernames:

@kainjow

This might be simply due to a missing "GitHub Name" entry in one's b.p.o account settings. This is necessary for legal reasons before we can look at this contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.




You can [check yourself](https://check-python-cla.herokuapp.com/) to see if the CLA has been received.

Thanks again for the contribution, we look forward to reviewing it!

File: Doc/library/winreg.rst [hunk 1]
```diff
@@ -443,7 +443,7 @@ This module offers the following functions:
    :const:`REG_SZ`, meaning only strings are supported.  Use the :func:`SetValueEx`
    function for support for other data types.
 
-   *value* is a string that specifies the new value.
+   *value* is a string or integer that specifies the new value.
```
Comment (serhiy-storchaka) [inline Doc/library/winreg.rst]: It can only be a string.

File: Doc/library/winreg.rst [hunk 2]
```diff
@@ -473,7 +473,7 @@ This module offers the following functions:
    *type* is an integer that specifies the type of the data. See
    :ref:`Value Types <value-types>` for the available types.
 
-   *value* is a string that specifies the new value.
+   *value* is a string or integer that specifies the new value.
```
Comment (serhiy-storchaka) [inline Doc/library/winreg.rst:476]: It can be not only a string or integer. Acceptable types depend on the *type* argument.

Comment (zooba) [reply Doc/library/winreg.rst:476]: Specifically None, str, int, list (of str) or implements the buffer protocol.

File: Doc/library/winreg.rst (hunk 1 above)
Comment (zooba) [reply Doc/library/winreg.rst]:
```suggestion
   *value* is a string that specifies the new value.
```

Correct

File: Doc/library/winreg.rst (hunk 2 above)
Comment (serhiy-storchaka) [reply Doc/library/winreg.rst:476]: And it is better to document what types are acceptable for specific value of the *type* argument in the corresponding section `value-types`.

**Pull request closed without merging**